python/ray/tune/impl/tuner_internal.py

```diff
@@ -191,8 +198,79 @@ def _maybe_warn_resource_contention(self):
                 stacklevel=4,
             )
 
+    def _validate_overwrite_trainable(self, original_trainable, overwrite_trainable):
```

Comment:
Love how thorough this is in providing a great user experience!

One point of caution is that this now introduces a new entire step for validating trainables, and I'm wondering if it might introduce new rough edges. 

Would it make sense to have a backdoor environment variable (e.g. `SKIP_OVEWRITE_TRAINABLE_VALIDATION`) that can be used to (1) skip this validation call and (2) completely skip deserializing the pickled trainable, when a trainable is directly passed in?  

cc @amogkam @Yard1 for thoughts as well.

Reply:
My main reason for this validation was to limit the usage as much as possible to just allow passing in the same trainable as before (only able to easily check for same name, same type). Tune also implicitly expects the trainable to be registered under the same name, so overwriting with a function/class that has a different name would error because the trainable would be saved under a different key - the error thrown in this case is pretty hard for a user to debug.